src/main/java/com/snowflake/kafka/connector/internal/SnowflakeIngestionServiceV1.java

```diff
@@ -44,24 +49,61 @@ public class SnowflakeIngestionServiceV1 implements SnowflakeIngestionService {
       String stageName,
       String pipeName,
       PrivateKey privateKey,
-      String userAgentSuffix) {
+      String userAgentSuffix,
+      @Nullable SnowflakeTelemetryService telemetry) {
+
+    this(
+        stageName,
+        pipeName,
+        createOrThrow(
+            accountName,
+            userName,
+            host,
+            port,
+            connectionScheme,
+            pipeName,
+            privateKey,
+            userAgentSuffix,
+            telemetry),
+        telemetry);
+    LOGGER.info("initialized the pipe connector for pipe {}", pipeName);
+  }
+
+  @VisibleForTesting
+  SnowflakeIngestionServiceV1(
+      String stageName,
+      String pipeName,
+      SimpleIngestManager ingestManager,
+      @Nullable SnowflakeTelemetryService telemetry) {
     this.stageName = stageName;
     this.pipeName = pipeName;
+    this.ingestManager = ingestManager;
+    setTelemetry(telemetry);
```

Comment:
Why use setter in the constructor?

Reply:
it was configured using builder - i've just reused the setter. i've updated the code to directly assign value.